<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.21">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>research</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="research_files/libs/clipboard/clipboard.min.js"></script>
<script src="research_files/libs/quarto-html/quarto.js" type="module"></script>
<script src="research_files/libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="research_files/libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="research_files/libs/quarto-html/popper.min.js"></script>
<script src="research_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="research_files/libs/quarto-html/anchor.min.js"></script>
<link href="research_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="research_files/libs/quarto-html/quarto-syntax-highlighting-ea1d7ac60288e0f1efdbc993fd8432ae.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="research_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="research_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="research_files/libs/bootstrap/bootstrap-9e3ffae467580fdb927a41352e75a2e0.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="research_files/libs/quarto-contrib/intersection-observer-polyfill-1.0.0/intersection-observer.js"></script>
<script src="research_files/libs/quarto-contrib/scrollama-3.2.0/scrollama.min.js"></script>
<script src="research_files/libs/quarto-contrib/closeread-0.1.0/closeread.js"></script>
<meta cr-debug-mode="false">
<meta cr-remove-header-space="true">


<link rel="stylesheet" href="_extensions/qmd-lab/closeread/closeread.css">
<link rel="stylesheet" href="styles.css">
</head>

<body class="quarto-light">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-full">

<main class="content column-page" id="quarto-document-content">




<section id="research-projects" class="level1">
<h1>Research Projects</h1>
<section id="metabolomic-analysis-of-gut-metabolites-in-colorectal-cancer-patients-crc" class="level3">
<h3 class="anchored" data-anchor-id="metabolomic-analysis-of-gut-metabolites-in-colorectal-cancer-patients-crc"><a href="https://link.springer.com/article/10.1007/s00726-025-03448-3">Metabolomic Analysis of Gut Metabolites in Colorectal Cancer Patients (CRC)</a></h3>
<section id="objectives" class="level5">
<h5 class="anchored" data-anchor-id="objectives">Objectives:</h5>
<p>Identify metabolite signatures and network-level metabolic disruptions distinguishing CRC patients from healthy controls.</p>
</section>
<section id="hurdles" class="level5">
<h5 class="anchored" data-anchor-id="hurdles">Hurdles:</h5>
<p>Low data coverage and a limited number of differentially expressed metabolites, making it difficult to identify robust discriminative signatures.</p>
</section>
<section id="rationale" class="level5">
<h5 class="anchored" data-anchor-id="rationale">Rationale:</h5>
<p>Metabolic alterations occur in coordinated patterns. By capturing these patterns through co-expressed modules correlated with disease status and exploring knowledge-based metabolite–metabolite interactions within these modules, it is possible to identify biomarkers and key metabolic mediators in CRC.</p>
</section>
<div id="R-box">
<section id="framework" class="level5">
<h5 class="anchored" data-anchor-id="framework">Framework:</h5>
<ul>
<li>Implemented an end-to-end pipeline, including raw data preprocessing, PQN normalization, differential expression analysis, WGCNA for module discovery, and eigengene differentiation for CRC-specific modules.</li>
<li>Applied intramodular connectivity to identify hub metabolites.</li>
<li>Applied PCA/PLS-DA for dimensionality reduction and feature prioritization.
</li></ul></section></div>

</section>
<section id="outcome" class="level5">
<h5 class="anchored">Outcome:</h5>
<p>Defined metabolite modules linked to CRC phenotypes, highlighted hub metabolites as candidate biomarkers, and revealed dysregulated metabolic networks and pathways.</p>
<div class="plot-row">
<div class="plot">
<h4 class="anchored" data-anchor-id="outcome">
TOM Heatmap
</h4>
<p><img src="plots/crc_TOM_plot.png"></p>
</div>
<div class="plot">
<h4 class="anchored">
PLS-DA VIP Features
</h4>
<p><img src="plots/crc_vip_plot2.png"></p>
</div>
<div class="plot">
<h4 class="anchored">
Enrichment Analysis
</h4>
<p><img src="plots/crc_msea_shared.jpg"></p>
</div>
</div>
<section id="find-more" class="level6">
<h6 class="anchored" data-anchor-id="find-more">Find More</h6>
<div id="L-box">
<strong><a href="https://link.springer.com/article/10.1007/s00726-025-03448-3">Paper Link</a></strong><br>
<strong><a href="https://bits-and-bites.netlify.app/">Source Code</a></strong>
</div>
<hr>
</section>
</section>
<section id="integrative-multi-omics-profiling-of-rhabdomyosarcoma-subtypes-reveals-distinct-molecular-pathways-and-biomarker-signatures" class="level3">
<h3 class="anchored" data-anchor-id="integrative-multi-omics-profiling-of-rhabdomyosarcoma-subtypes-reveals-distinct-molecular-pathways-and-biomarker-signatures"><strong><a href="https://www.mdpi.com/2073-4409/14/14/1115">Integrative Multi-Omics Profiling of Rhabdomyosarcoma Subtypes Reveals Distinct Molecular Pathways and Biomarker Signatures</a></strong></h3>
<section id="objectives-1" class="level5">
<h5 class="anchored" data-anchor-id="objectives-1">Objectives:</h5>
<p>Uncover subtype-specific molecular signatures and key drivers of RMS using integrative proteomics and metabolomics.</p>
</section>
<section id="hurdles-1" class="level5">
<h5 class="anchored" data-anchor-id="hurdles-1">Hurdles:</h5>
<p>High heterogeneity across subtypes and difficulty distinguishing biomarkers specific to each subtype.</p>
</section>
<section id="rationale-1" class="level5">
<h5 class="anchored" data-anchor-id="rationale-1">Rationale:</h5>
<p>Rather than focusing on individual genes, identifying altered networks and their central regulators provides a more comprehensive understanding of subtype-specific etiologies and therapeutic targets.</p>
</section>
<div id="R-box">
<section id="framework-1" class="level5">
<h5 class="anchored" data-anchor-id="framework-1">Framework:</h5>
<ul>
<li>Applied WGCNA to identify co-expressed modules and tested eigengene differentiation using Wilcoxon statistics.</li>
<li>Constructed PPI networks integrating hub and differentially expressed proteins, then applied centrality analysis to identify key drivers.
</li></ul></section></div>

</section>
<section id="outcome-1" class="level5">
<h5 class="anchored">Outcome:</h5>
<p>Characterized subtype-specific molecular landscapes in RMS, pinpointed central regulators, and mapped key altered pathways.</p>
<div class="plot-row">
<div class="plot">
<h4 class="anchored" data-anchor-id="outcome-1">
Module-phenotype Heatmap
</h4>
<p><img src="plots/rhabdo_WGCNA.png"></p>
</div>
<div class="plot">
<h4 class="anchored">
Module Eigengene Barplot
</h4>
<p><img src="plots/rhabdo_modules_discrimination.png"></p>
</div>
</div>
<section id="find-more-1" class="level6">
<h6 class="anchored" data-anchor-id="find-more-1">Find More</h6>
<div id="L-box">
<strong><a href="https://www.mdpi.com/2073-4409/14/14/1115">Paper Link</a></strong><br>
<strong><a href="https://github.com/Menna-Arafat/Rhabdomyosarcome_project">Source Code</a></strong>
</div>
<hr>
</section>
</section>
<section id="crosstalkx-a-mutual-informationbased-framework-for-pathway-interaction-inference" class="level3">
<h3 class="anchored" data-anchor-id="crosstalkx-a-mutual-informationbased-framework-for-pathway-interaction-inference">CrosstalkX: A Mutual Information–Based Framework for Pathway Interaction Inference</h3>
<section id="objectives-2" class="level5">
<h5 class="anchored" data-anchor-id="objectives-2">Objectives:</h5>
<p>Develop a computational tool to detect nonlinear dependencies and interactions among signaling pathways and transcription factors (TFs).</p>
</section>
<section id="hurdles-2" class="level5">
<h5 class="anchored" data-anchor-id="hurdles-2">Hurdles:</h5>
<p>Traditional correlation-based approaches overlook nonlinear associations; scalable and robust statistical modeling was required.</p>
</section>
<section id="rationale-2" class="level5">
<h5 class="anchored" data-anchor-id="rationale-2">Rationale:</h5>
<p>Pathways function through coordinated interactions rather than isolated activity. We applied a mutual information algorithm between pathway gene sets with prior PPI constraints, inferring edges only when significant mutual information was detected between first-degree interacting genes. Mutual information was then inferred at the pathway level, with randomization to assess significance. Significant interactions reveal functional interplay underlying complex diseases.</p>
</section>
<div id="R-box">
<section id="framework-2" class="level5">
<h5 class="anchored" data-anchor-id="framework-2">Framework:</h5>
<ul>
<li>Developed a mutual information–based inference algorithm in R to construct pathway–pathway and pathway–TF interaction networks.</li>
<li>Core pathways and TFs were prioritized by interaction degree.
</li></ul></section></div>

</section>
<section id="outcome-2" class="level5">
<h5 class="anchored">Outcome:</h5>
<p>Developed CrosstalkX, an R package for mutual information–based pathway interaction analysis, enabling the detection of nonlinear dependencies between pathway gene sets. The tool identifies highly interacting signaling pathways and prioritizes core pathways with high interaction degrees, revealing key functional Interplay that potentially coordinate critical biological processes.</p>
<div class="plot-row">
<div class="plot">
<h4 class="anchored" data-anchor-id="outcome-2">
Enrichment Plot
</h4>
<p><img src="plots/crosstalkx_enrich.jpg"></p>
</div>
<div class="plot">
<h4 class="anchored">
Crosstalk Plot
</h4>
<p><img src="plots/crosstalkx_crossTalk.png"></p>
</div>
<div class="plot">
<h4 class="anchored">
TF-target Plot
</h4>
<p><img src="plots/crosstalkx_TF_target.png"></p>
</div>
</div>
<section id="find-more-2" class="level6">
<h6 class="anchored" data-anchor-id="find-more-2">Find More</h6>
<div id="L-box">
<p><strong><a href="https://github.com/Menna-Arafat/CrosstalkX">Source Code</a></strong></p>
</div>
<hr>
</section>
</section>
<section id="a-multi-omics-and-mutual-information-framework-to-decode-pathway-crosstalk-and-therapeutic-vulnerabilities-in-tnbc" class="level3">
<h3 class="anchored" data-anchor-id="a-multi-omics-and-mutual-information-framework-to-decode-pathway-crosstalk-and-therapeutic-vulnerabilities-in-tnbc">A Multi-Omics and Mutual Information Framework to Decode Pathway Crosstalk and Therapeutic Vulnerabilities in TNBC</h3>
<section id="objectives-3" class="level5">
<h5 class="anchored" data-anchor-id="objectives-3">Objectives:</h5>
<p>Prioritize therapeutic targets and repurpose drugs for TNBC by integrating multi-omics data and pathway crosstalk analysis.</p>
</section>
<section id="hurdles-3" class="level5">
<h5 class="anchored" data-anchor-id="hurdles-3">Hurdles:</h5>
<p>Transcriptome data were highly noisy, with enrichment analysis yielding nonspecific results. Integration across heterogeneous omics layers made it difficult to distinguish meaningful pathway interactions from background noise.</p>
</section>
<section id="rationale-3" class="level5">
<h5 class="anchored" data-anchor-id="rationale-3">Rationale:</h5>
<p>Multi-omics integration via MOFA, with ARD-induced sparsity, reduces noise by setting weights of non-contributing genes close to zero. TF-target inference applied to MOFA top features, combined with univariate linear modeling (decoupleR), identifies active TFs. Centrality algorithms prioritize key targets, while DepMap essentiality scores validate lethality. Mutual information–based crosstalk analysis could reveal core interacting pathways. Finally, Macau-based sensitivity analysis links pathway states to drug responsiveness, exposing actionable vulnerabilities.</p>
</section>
<div id="R-box">
<section id="framework-3" class="level5">
<h5 class="anchored" data-anchor-id="framework-3">Framework:</h5>
<ul>
<li>Retrieved six-layer TCGA data, applied MOFA for latent factor inference, and extracted top features.</li>
<li>Inferred TF activity with decoupleR, applied centrality analysis, and introduced mutual information–based crosstalk.</li>
<li>Linked central genes and pathways to candidate drugs via CMAP, applied network synergy prediction.</li>
<li>Validated essential targets with DepMap, and incorporated survival analysis.</li>
<li>Implemented Macau for target sensitivity prediction.
</li></ul></section></div>

</section>
<section id="outcome-3" class="level5">
<h5 class="anchored">Outcome:</h5>
<p>Identified core interacting pathways, prioritized therapeutic targets, and refined drug combinations aligned with TNBC molecular states.</p>
<div class="plot-row">
<div class="plot">
<h4 class="anchored" data-anchor-id="outcome-3">
Crosstalk Plot
</h4>
<p><img src="plots/tnbc_crossTalk_MI.png"></p>
</div>
<div class="plot">
<h4 class="anchored">
Chord Plot
</h4>
<p><img src="plots/tnbc_chord.png"></p>
</div>
<div class="plot">
<h4 class="anchored">
Heatmap Plot
</h4>
<p><img src="plots/tnbc_heatmaps_side_by_side.png"></p>
</div>
</div>
<section id="find-more-3" class="level6">
<h6 class="anchored" data-anchor-id="find-more-3">Find More</h6>
<div id="L-box">
<p><strong><a href="https://github.com/Menna-Arafat/TNBC_project">Source Code</a></strong></p>
</div>
<hr>
</section>
</section>
<section id="enhanced-bone-regeneration-in-rat-gingival-mesenchymal-stem-cells" class="level3">
<h3 class="anchored" data-anchor-id="enhanced-bone-regeneration-in-rat-gingival-mesenchymal-stem-cells">Enhanced Bone Regeneration in Rat Gingival Mesenchymal Stem Cells</h3>
<section id="objectives-4" class="level5">
<h5 class="anchored" data-anchor-id="objectives-4">Objectives:</h5>
<p>Characterize molecular programs and cellular contributions driving bone regeneration under diverse interventions.</p>
</section>
<section id="hurdles-4" class="level5">
<h5 class="anchored" data-anchor-id="hurdles-4">Hurdles:</h5>
<p>Multi-layer temporal datasets across experimental settings with three time-point measurements each, requiring disentanglement of heterogeneous cell-type contributions.</p>
</section>
<section id="rationale-4" class="level5">
<h5 class="anchored" data-anchor-id="rationale-4">Rationale:</h5>
<p>Factor analysis of spatio-temporal profiles can uncover latent factors that covary across time points and explain differentiation trajectories. Understanding these regulatory programs will optimize scaffold- and laser-based regenerative therapies.</p>
</section>
<div id="R-box">
<section id="framework-4" class="level5">
<h5 class="anchored" data-anchor-id="framework-4">Framework:</h5>
<ul>
<li>Applied MEFISTO for time-resolved integration of protein expression data.</li>
<li>Inferred TF activity with decoupleR, explored pathway crosstalk.</li>
<li>Applied CIBERSORT deconvolution against a dental cell atlas to map protein changes to cell types.
</li></ul></section></div>

</section>
<section id="outcome-4" class="level5">
<h5 class="anchored">Outcome:</h5>
<p>Revealed time-coordinated regulatory programs, identified TF–pathway axes, and mapped cell-type contributions enhancing regenerative efficacy.</p>
<div class="plot-row">
<div class="plot">
<h4 class="anchored" data-anchor-id="outcome-4">
Upset Plot
</h4>
<p><img src="plots/gmsc_upsetplot.png"></p>
</div>
<div class="plot">
<h4 class="anchored">
Latent Factor1 Plot
</h4>
<p><img src="plots/gmsc_MEFISTO_factor1.png"></p>
</div>
<div class="plot">
<h4 class="anchored">
Stacked Barplot for SC Deconvolution
</h4>
<p><img src="plots/gmsc_Cibersort.png"></p>
</div>
<div class="plot">
<h4 class="anchored">
TF-Pathway Crosstalk Plot
</h4>
<p><img src="plots/gmsc_crossTalk.png"></p>
</div>
</div>
<section id="find-more-4" class="level6">
<h6 class="anchored" data-anchor-id="find-more-4">Find More</h6>
<div id="L-box">
<p><strong><a href="https://github.com/Menna-Arafat/GMSC_project">Source Code</a></strong></p>
</div>
<hr>
</section>
</section>
<section id="integrative-analysis-of-extracellular-vesicle-mirnas-reveals-molecular-signatures-and-regulatory-networks-in-breast-cancer-subtypes" class="level3">
<h3 class="anchored" data-anchor-id="integrative-analysis-of-extracellular-vesicle-mirnas-reveals-molecular-signatures-and-regulatory-networks-in-breast-cancer-subtypes">Integrative Analysis of Extracellular Vesicle miRNAs Reveals Molecular Signatures and Regulatory Networks in Breast Cancer Subtypes</h3>
<section id="objectives-5" class="level5">
<h5 class="anchored" data-anchor-id="objectives-5">Objectives:</h5>
<p>Discover extracellular vesicle miRNA signatures and regulatory networks distinguishing breast cancer subtypes.</p>
</section>
<section id="hurdles-5" class="level5">
<h5 class="anchored" data-anchor-id="hurdles-5">Hurdles:</h5>
<p>Batch variability across multiple datasets and low coverage for some studies and missing metadata for others.</p>
</section>
<section id="rationale-5" class="level5">
<h5 class="anchored" data-anchor-id="rationale-5">Rationale:</h5>
<p>Batch effect correction using limma voom could help git rid of wanted source of variability. MINT-sPLSDA could offer a robust way of integrating data through harmonizing data using z scoring and then allow us to choose the best number of components to integrate the datasets that allow better classification between groups. miRNA does not work individually usually more than miRNA targeting the same gene change together and thus we could prioritize differentially expressed miRNA for each subtypes by setting the condition that at least 3 mirna targeting the same genes should be differentially expressed to infer translational molecular alteration.</p>
</section>
<div id="R-box">
<section id="framework-5" class="level5">
<h5 class="anchored" data-anchor-id="framework-5">Framework:</h5>
<ul>
<li>Preprocessed and batch-corrected datasets.</li>
<li>Applied MINT-sPLS-DA for multivariate integration and subtype classification.</li>
<li>Constructed miRNA–target networks requiring triply supported interactions, and performed enrichment analyses.
</li></ul></section></div>

</section>
<section id="outcome-5" class="level5">
<h5 class="anchored">Outcome:</h5>
<p>Identified discriminative miRNA panels for subtype classification, uncovered key pathways under vesicle-miRNA regulation, and proposed subtype-specific molecular biomarkers and potential molecularly altered networks.</p>
<div class="plot-row">
<div class="plot">
<h4 class="anchored" data-anchor-id="outcome-5">
Heatmap for BC vs Ctrl
</h4>
<p><img src="plots/mirna_BC_vs_Ctrlheatmap.png"></p>
</div>
<div class="plot">
<h4 class="anchored">
ROC Curve for BC vs Ctrl
</h4>
<p><img src="plots/mirna_BC_vs_CtrlROC_plot.png"></p>
</div>
<div class="plot">
<h4 class="anchored">
Heatmap for BC Subtypes
</h4>
<p><img src="plots/mirna_subtypes_heatmap.png"></p>
</div>
<div class="plot">
<h4 class="anchored">
ROC Curve for BC Subtypes
</h4>
<p><img src="plots/mirna_subtypes_ROC_plot.png"></p>
</div>
</div>
<section id="find-more-5" class="level6">
<h6 class="anchored" data-anchor-id="find-more-5">Find More</h6>
<div id="L-box">
<p><strong><a href="https://github.com/Menna-Arafat/BC_EVs_miRNA">Source Code</a></strong></p>
</div>
<hr>
</section>
</section>
<section id="proteomic-profiling-of-pancreatic-cancer-patients-time-series-study-before-and-after-whipple-surgery" class="level3">
<h3 class="anchored" data-anchor-id="proteomic-profiling-of-pancreatic-cancer-patients-time-series-study-before-and-after-whipple-surgery">Proteomic Profiling of Pancreatic Cancer Patients: Time Series Study Before and After Whipple Surgery</h3>
<section id="objectives-6" class="level5">
<h5 class="anchored" data-anchor-id="objectives-6">Objectives:</h5>
<p>Track temporal protein dynamics and functional reorganization in pancreatic cancer patients undergoing Whipple surgery without developing diabetic complications.</p>
</section>
<section id="hurdles-6" class="level5">
<h5 class="anchored" data-anchor-id="hurdles-6">Hurdles:</h5>
<p>Identifying signatures correlated with long-term glycemic stability rather than short-term post-surgical responses.</p>
</section>
<section id="rationale-6" class="level5">
<h5 class="anchored" data-anchor-id="rationale-6">Rationale:</h5>
<p>Surgery induces systemic proteomic changes. Mapping these dynamics reveals stress responses, repair mechanisms, and processes contributing to stable glycemic control.</p>
</section>
<section id="framework-6" class="level5">
<h5 class="anchored" data-anchor-id="framework-6">Framework:</h5>
<p>Applied WGCNA to detect co-expression modules and correlated eigengenes with time metadata. For the module associated with recovery (time point 3), modeled temporal changes using GAMs with patient-specific intercepts. Identified hub proteins (datKME &gt; 0.85), integrated with STRING-based PPI networks, applied Louvain clustering, and performed enrichment analysis.</p>
</section>
<section id="outcome-6" class="level5">
<h5 class="anchored">Outcome:</h5>
<p>Highlighted dynamic proteomic modules, identified central hub proteins in recovery, and revealed pathway modules with coordinated activity and crosstalk relevant to post-surgical adaptation.</p>
<div class="plot-row">
<div class="plot">
<h4 class="anchored" data-anchor-id="outcome-6">
Module-phenotype Heatmap
</h4>
<p><img src="plots/pancreas_heat2_module_trait_WGCNA.png"></p>
</div>
<div class="plot">
<h4 class="anchored">
GAM Plot
</h4>
<p><img src="plots/pancreas_gam_plot.png"></p>
</div>
<div class="plot">
<h4 class="anchored">
PPI network &amp; Louvain Detected Communities
</h4>
<p><img src="plots/pancreas_communities_ppi_tf_blue.png"></p>
</div>
<div class="plot">
<h4 class="anchored">
Enriched Pathways in Blue Module
</h4>
<p><img src="plots/pancreas_pathway_structural_similarity_blue_last.png"></p>
</div>
</div>
<section id="find-more-6" class="level6">
<h6 class="anchored" data-anchor-id="find-more-6">Find More</h6>
<div id="L-box">
<p><strong><a href="https://github.com/Menna-Arafat/Whiple_surgery_project">Source Code</a></strong></p>
</div>
<hr>
</section>
</section>
</section>
</section></main></div>














<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
 <!-- /content -->




</body></html>